docs/src/content/docs/fr/reference/plugins.md

```diff
@@ -0,0 +1,162 @@
+
+Pour étendre une option de configuration existante sans la remplacer, étendez la valeur existante dans votre nouvelle valeur.
+Dans l'exemple suivant, un nouveau compte de média [`social`](/fr/reference/configuration/#social) est ajouté à la configuration existante en étendant `config.social` dans le nouvel objet `social` :
+
+```ts {6-11}
+// module-extension.ts
+export default {
+  name: 'ajout-twitter-plugin',
+  hooks: {
+    setup({ config, updateConfig }) {
+      updateConfig({
+        social: {
+          ...config.social,
+          twitter: 'https://twitter.com/astrodotbuild',
+        },
+      });
+    },
+  },
+};
+```
+
+#### `addIntegration`
+
+**Type :** `(integration: AstroIntegration) => void`
+
+Une fonction de rappel pour ajouter une [intégration Astro](https://docs.astro.build/fr/reference/integrations-reference/) requise par le module d'extension.
+
+Dans l'exemple suivant, le module d'extension vérifie d'abord si [l'intégration React d'Astro](https://docs.astro.build/fr/guides/integrations-guide/react/) est configurée et, si ce n'est pas le cas, utilise `addIntegration()` pour l'ajouter :
+
+```ts {14} "addIntegration,"
+// module-extension.ts
+import react from '@astrojs/react';
+
+export default {
+  name: 'plugin-utilisant-react',
+  hooks: {
+    plugin({ addIntegration, astroConfig }) {
+      const isReactLoaded = astroConfig.integrations.find(
+        ({ name }) => name === '@astrojs/react'
+      );
+
+      // Ajoute seulement l'intégration React si elle n'est pas déjà chargée.
+      if (!isReactLoaded) {
+        addIntegration(react());
+      }
+    },
+  },
+};
+```
+
+#### `astroConfig`
+
+**Type :** `AstroConfig`
+
+Une copie en lecture seule de la [configuration d'Astro](https://docs.astro.build/fr/reference/configuration-reference/) fournie par l'utilisateur.
+
+#### `command`
+
+**Type :** `'dev' | 'build' | 'preview'`
+
+La commande utilisée pour exécuter Starlight :
+
+- `dev` - Le projet est exécuté avec `astro dev`
+- `build` - Le projet est exécuté avec `astro build`
+- `preview` - Le projet est exécuté avec `astro preview`
+
+#### `isRestart`
+
+**Type :** `boolean`
+
+`false` lorsque le serveur de développement démarre, `true` lorsqu'un rechargement est déclenché.
+
+#### `logger`
+
+**Type :** `AstroIntegrationLogger`
+
+Une instance du [journaliseur (logger) d'intégration Astro](https://docs.astro.build/fr/reference/integrations-reference/#astrointegrationlogger) que vous pouvez utiliser pour écrire des messages de journalisation.
+Tous les messages seront préfixés par le nom du module d'extension.
+
+```ts {6}
+// module-extension.ts
+export default {
+  name: 'long-processus-plugin',
+  hooks: {
+    plugin({ logger }) {
+      logger.info("Démarrage d'un long processus…");
+      // Un long processus…
+    },
+  },
+};
+```
+
+L'exemple ci-dessus affichera un message qui inclut le message d'information fourni :
+
+```plaintext frame="terminal"
```

Comment:
This change is to avoid weird highlighting due to the `'` character:

<img width="742" alt="image" src="https://github.com/withastro/starlight/assets/494699/b832edab-d050-4a26-a5fc-da9b5db96ef8">